File: PDOO/p4/Civitas/src/civitas/CasillaSorpresa.java

package civitas;

import java.util.ArrayList;

public class CasillaSorpresa extends Casilla {
    private MazoSorpresas mazo;
    private Sorpresa sorpresa;
    CasillaSorpresa(MazoSorpresas mazo, String nombre){
        //Para crear la casilla de tipo sorpresa
        super(nombre);
        this.mazo = mazo;
    }
    
    @Override
    public void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if (super.jugadorCorrecto(iactual, todos)){
            sorpresa = mazo.siguiente();
            super.informe(iactual, todos);
            sorpresa.aplicarAJugador(iactual, todos);
        }
    }
    
    @Override
    public String toString(){
        return "Casilla{ nombre=" + super.getNombre() + " }";
    }
}

File: PDOO/p4/civitas_rb/lib/casilla_sorpresa.rb
module Civitas

class Casilla_sorpresa < Casilla
  
  def initialize(mazo, nombre)
    super(nombre)
    @mazo=mazo
  end
  
  def recibejugador(iactual, todos)
    if(jugador_correcto(iactual, todos))
      @sorpresa= @mazo.siguiente
      informe(iactual, todos)
      @sorpresa.aplicar_a_jugador(iactual, todos)
    end
  end
  
  def to_string()
    return "#{super}"
  end
  
end
end